### ISSUE TITLE
CKV_AWS_300: false-positive in case of disabled rule in dynamic block

### ISSUE THREAD

**Issue by kiss-less** [bridgecrewio/checkov#6028]
**Describe the issue**
The CKV_AWS_300 fails if there is a disabled dynamic rule block with `abort_incomplete_multipart_upload`, even if the `abort_incomplete_multipart_upload` has been already defined in another rule.

**Examples**
FYI: the below configuration has been intentionally simplified to just showcase what's important for the issue.

Check passes with the following config:
```
resource "aws_s3_bucket_lifecycle_configuration" "bucket_lifecycle" {
  bucket = aws_s3_bucket.bucket.id

  rule {
    id     = "AbortIncompleteMultipartUploadEnabled"
    status = "Enabled"

    abort_incomplete_multipart_upload {
      days_after_initiation = 7
    }
  }
  dynamic "rule" {
    for_each = var.bucket_transition_lifecycle_rule
    content {
      id     = rule.value["id"]
      status = "Enabled"

      abort_incomplete_multipart_upload {
        days_after_initiation = 7
      }
    }
  }
}
```
Check fails with the same config if the dynamic rule is disabled:
```
resource "aws_s3_bucket_lifecycle_configuration" "bucket_lifecycle" {
  bucket = aws_s3_bucket.bucket.id

  rule {
    id     = "AbortIncompleteMultipartUploadEnabled"
    status = "Enabled"

    abort_incomplete_multipart_upload {
      days_after_initiation = 7
    }
  }
  dynamic "rule" {
    for_each = var.bucket_transition_lifecycle_rule
    content {
      id     = rule.value["id"]
      status = "Disabled"

      abort_incomplete_multipart_upload {
        days_after_initiation = 7
      }
    }
  }
}
```

**Version (please complete the following information):**
 - Checkov Version 3.2.20

**Comment by abondrndc** [comment]
Hey @kiss-less does this still result in a failure for you? Using your examples the check passes on version 3.2.23.

You state the example is intentionally simplified, does the failing resource include a `filter`?

**Comment by kiss-less** [comment]
Hey @abondrndc thank you for taking a look into the problem! Sorry, I just realized that I haven't provided the `variables.tf` with the value of `var.bucket_transition_lifecycle_rule`, so please find below the updated examples that include the variable:

### Config that goes through:

test.tf:
```
resource "aws_s3_bucket_lifecycle_configuration" "bucket_lifecycle" {
  bucket = aws_s3_bucket.bucket.id

  rule {
    id     = "AbortIncompleteMultipartUploadEnabled"
    status = "Enabled"

    abort_incomplete_multipart_upload {
      days_after_initiation = 7
    }
  }
  dynamic "rule" {
    for_each = var.bucket_transition_lifecycle_rule
    content {
      id     = rule.value["id"]
      status = rule.value["status"]

      abort_incomplete_multipart_upload {
        days_after_initiation = rule.value["abort_incomplete_multipart_upload_days"]
      }
    }
  }
}
```
variables.tf:
```
variable "bucket_transition_lifecycle_rule" {
  type        = list(map(any))
  description = "Set-to-ia lifecycle rule."
  default = [{
    "status"                                 = "Enabled"
    "id"                                     = "set-to-ia"
    "abort_incomplete_multipart_upload_days" = 7
  }]
}
```
![image](https://github.com/bridgecrewio/checkov/assets/20906885/f4db42e8-ffe8-4596-bbb1-c9588eb38533)


### Config that doesn't go through:

test.tf:
```
resource "aws_s3_bucket_lifecycle_configuration" "bucket_lifecycle" {
  bucket = aws_s3_bucket.bucket.id

  rule {
    id     = "AbortIncompleteMultipartUploadEnabled"
    status = "Enabled"

    abort_incomplete_multipart_upload {
      days_after_initiation = 7
    }
  }
  dynamic "rule" {
    for_each = var.bucket_transition_lifecycle_rule
    content {
      id     = rule.value["id"]
      status = rule.value["status"]

      abort_incomplete_multipart_upload {
        days_after_initiation = rule.value["abort_incomplete_multipart_upload_days"]
      }
    }
  }
}
```
variables.tf:
```
variable "bucket_transition_lifecycle_rule" {
  type        = list(map(any))
  description = "Set-to-ia lifecycle rule."
  default = [{
    "status"                                 = "Disabled"
    "id"                                     = "set-to-ia"
    "abort_incomplete_multipart_upload_days" = 7
  }]
}
```
![image](https://github.com/bridgecrewio/checkov/assets/20906885/5460057f-aa85-46a7-8bf6-d250daf39e84)

The only difference between them is the value of the var.bucket_transition_lifecycle_rule["status"]. As you can see in the screenshot, the check fails with the latter config even if I use the latest version of checkov as of now (3.2.24).

By "intentionally simplified" I mean that I only included the necessary properties to showcase the problem, there is no `filter` in the failing resource.

P.S. steps to reproduce:

1. Create test.tf and variables.tf with the above contents
2. docker run --rm -it -v ./:/app --entrypoint /bin/bash bridgecrew/checkov:3.2.24
3. cd /app && checkov -d .

**Comment by abondrndc** [comment]
I'm still learning how checkov rules are developed, but the logic that is evaluating the rules doesn't see the enabled rule.

Here is this `conf` value being passed into the CKV_AWS_300.

```json
{
    "__end_line__": 33,
    "__start_line__": 11,
    "bucket": [
        "test"
    ],
    "dynamic": [
        {
            "rule": {
                "content": [
                    {
                        "abort_incomplete_multipart_upload": [
                            {
                                "days_after_initiation": [
                                    "rule.value[\"abort_incomplete_multipart_upload_days\"]"
                                ]
                            }
                        ],
                        "id": [
                            "rule.value[\"id\"]"
                        ],
                        "status": [
                            "rule.value[\"status\"]"
                        ]
                    }
                ],
                "for_each": [
                    {
                        "abort_incomplete_multipart_upload_days": [
                            7
                        ],
                        "id": [
                            "set-to-ia"
                        ],
                        "status": [
                            "Disabled"
                        ]
                    }
                ]
            }
        }
    ],
    "rule": [
        {
            "abort_incomplete_multipart_upload": {
                "days_after_initiation": [
                    7
                ]
            },
            "id": [
                "set-to-ia"
            ],
            "status": [
                "Disabled"
            ]
        },
        {
            "abort_incomplete_multipart_upload": [
                {
                    "days_after_initiation": [
                        "rule.value[\"abort_incomplete_multipart_upload_days\"]"
                    ]
                }
            ],
            "id": [
                "rule.value[\"id\"]"
            ],
            "status": [
                "rule.value[\"status\"]"
            ]
        },
        {
            "abort_incomplete_multipart_upload": [
                {
                    "days_after_initiation": [
                        "rule.value[\"abort_incomplete_multipart_upload_days\"]"
                    ]
                }
            ],
            "id": [
                "rule.value[\"id\"]"
            ],
            "status": [
                "rule.value[\"status\"]"
            ]
        }
    ],
    "__address__": "aws_s3_bucket_lifecycle_configuration.bucket_lifecycle"
}
```

**Comment by kiss-less** [comment]
Yep, that's clearly the root cause. I thought in the same direction as you actually: also added some verbosity to the check and got a pretty similar result:

![image](https://github.com/bridgecrewio/checkov/assets/20906885/432e954e-2598-4937-a549-b9da1454af29)
